engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/hydration/NadelBatchHydrationMatchStrategy.kt

```diff
@@ -1,8 +1,17 @@
 package graphql.nadel.enginekt.blueprint.hydration
 
+import graphql.nadel.enginekt.transform.query.NadelQueryPath
+
 sealed class NadelBatchHydrationMatchStrategy {
     object MatchIndex : NadelBatchHydrationMatchStrategy()
+
     data class MatchObjectIdentifier(
-        val objectId: String,
+        val sourceId: NadelQueryPath,
+        // todo should also be NadelQueryPath
+        val resultId: String,
+    ) : NadelBatchHydrationMatchStrategy()
+
+    data class MatchObjectIdentifiers(
+        val objectIds: List<MatchObjectIdentifier>,
```

Comment:
So the runtime representation of the complex identified by is just list of `MatchObjectIdentifier`

Also `MatchObjectIdentifier` has been updated with a `sourceId`. Example `.toString()` is

```
MatchObjectIdentifiers(
  objectIds = [
    {
      sourceId: ["confluencePage", "id"],
      resultId: "id",
    },
    {
      sourceId: ["confluencePage", "status"],
      resultId: "status",
    },
  ]
)
```